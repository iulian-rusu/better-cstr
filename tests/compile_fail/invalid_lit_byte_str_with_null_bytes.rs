use better_cstr::c;

fn main() {
    c!(b"This byte string contains a null byte \0!");
}
