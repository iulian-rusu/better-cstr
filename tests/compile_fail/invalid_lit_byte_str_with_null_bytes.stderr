error: proc macro panicked
 --> tests/compile_fail/invalid_lit_byte_str_with_null_bytes.rs:4:5
  |
4 |     c!(b"This byte string contains a null byte \0!");
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Byte string literal must be a valid C string: FromVecWithNulError { error_kind: InteriorNul(38), bytes: [84, 104, 105, 115, 32, 98, 121, 116, 101, 32, 115, 116, 114, 105, 110, 103, 32, 99, 111, 110, 116, 97, 105, 110, 115, 32, 97, 32, 110, 117, 108, 108, 32, 98, 121, 116, 101, 32, 0, 33, 0] }
